Wikipedia/Code/AppearanceSettingsViewController.swift

```diff
@@ -0,0 +1,245 @@
+import UIKit
+
+@objc public protocol WMFAppearanceSettingsViewControllerDelegate {
+    func themeCangedInController(_ controller: AppearanceSettingsViewController, theme: Theme)
+}
+
+protocol AppearanceSettingsItem {
+    var title: String? { get }
+}
+
+struct AppearanceSettingsSwitchItem: AppearanceSettingsItem {
+    let title: String?
+}
+
+struct AppearanceSettingsCheckmarkItem: AppearanceSettingsItem {
+    let title: String?
+    let theme: Theme
+    let checkmarkAction: () -> Void
+}
+
+struct AppearanceSettingsCustomViewItem: AppearanceSettingsItem {
+    let title: String?
+    let viewController: UIViewController
+}
+
+struct AppearanceSettingsSection {
+    let headerTitle: String
+    let footerText: String?
+    let items: [AppearanceSettingsItem]
+}
+
+@objc(WMFAppearanceSettingsViewController)
+open class AppearanceSettingsViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
+    
+    @IBOutlet weak var tableView: UITableView!
+    
+    var sections = [AppearanceSettingsSection]()
+    
+    fileprivate var theme = Theme.standard
+    
+    open weak var delegate: WMFAppearanceSettingsViewControllerDelegate?
+    
+    
+    static var disclosureText: String {
+        var text = WMFLocalizedString("default-theme-display-name", value: "Default", comment: "Default theme name presented to the user")
+        let currentAppTheme = UserDefaults.wmf_userDefaults().wmf_appTheme
+        
+        if currentAppTheme == Theme.sepia {
+            text = WMFLocalizedString("sepia-theme-display-name", value: "Sepia", comment: "Sepia theme name presented to the user")
+        } else if currentAppTheme == Theme.dark || currentAppTheme == Theme.darkDimmed {
+            text = WMFLocalizedString("dark-theme-display-name", value: "Dark", comment: "Dark theme name presented to the user")
```

Comment:
The localized theme names can be added as `localizedName` or `displayName` to the theme object.  That way, you can do `let text = currentAppTheme.localizedName`.